Added pyladies course for Olomouc spring 2025

mcurlej opened the pull request:
The compiled sources should be https://github.com/mcurlej/naucse-python/tree/compiled/spring2025-olomouc.
The git branch is https://github.com/mcurlej/naucse-python/tree/spring2025-olomouc.

WDYT?

@encukou @frenzymadness @Kobzol

Kobzol reviewed (commented):
Hmm, I guess that we should have either used the main branch + separate subdirectory for each run, or a per-run branch from the beginning. Now we should take care not to overwrite the main branch in the future :) Otherwise LGTM.

mcurlej commented:
Yeah i don't like the main branch. I updated also our winter course so it has its own branch. If there would only one main branch with the actual course we the older course would be also rewritten to the future course. With every course having their own branch this will never happen.
